openwisp_notifications/api/views.py

```diff
@@ -0,0 +1,67 @@
+import uuid
+
+from django.utils.translation import gettext_lazy as _
+from django_filters.rest_framework import DjangoFilterBackend
+from openwisp_notifications.api.serializers import NotificationSerializer
+from openwisp_notifications.swapper import load_model
+from rest_framework import status
+from rest_framework.authentication import BasicAuthentication, SessionAuthentication
+from rest_framework.generics import GenericAPIView
+from rest_framework.mixins import ListModelMixin
+from rest_framework.pagination import PageNumberPagination
+from rest_framework.permissions import DjangoModelPermissions
+from rest_framework.response import Response
+
+Notification = load_model('Notification')
+
+
+class NotificationPaginator(PageNumberPagination):
+    page_size = 10
+    page_size_query_param = 'page_size'
+    max_page_size = 100
+
+
+class BaseNotificationView(GenericAPIView):
+    model = Notification
+    authentication_classes = [SessionAuthentication, BasicAuthentication]
+    permission_classes = [DjangoModelPermissions]
+    queryset = Notification.objects.all()
+    serializer_class = NotificationSerializer
+    filter_backends = [DjangoFilterBackend]
+    filterset_fields = ['unread']
```

Comment:
I don't think `filter_backends` and `filterset_fields` are needed in `NotificationReadView` so they should be removed from the `BaseNotificationView and send to the `NotificatiobListView`

Reply:
Great catch!